src/Processors/QueryPlan/ReadFromRemote.cpp

```diff
@@ -437,16 +454,64 @@ void ReadFromParallelRemoteReplicasStep::initializePipeline(QueryPipelineBuilder
         shuffled_pool = shard.pool->getShuffledPools(current_settings, priority_func);
     }
 
-    for (size_t i=0; i < all_replicas_count; ++i)
+    std::vector<ConnectionPoolPtr> pools_to_use;
+    pools_to_use.reserve(shuffled_pool.size());
+    if (exclude_local_replica)
     {
-        IConnections::ReplicaInfo replica_info
+        std::vector<size_t> local_addr_possitions;
```

Comment:
this variable is not used it seems. 
could this be replaced with smth simpler like remove_if + erase?